geomag/world_magnetic_model.py

```diff
@@ -0,0 +1,294 @@
+                if m <= n:
+                    self.coefficient[m][n] = schmidt_norm[m][n] * self.coefficient[m][n]
+                    self.coefficient_dot[m][n] = schmidt_norm[m][n] * self.coefficient_dot[m][n]
+                else:
+                    self.coefficient[m][n] = schmidt_norm[n + 1][m] * self.coefficient[m][n]
+                    self.coefficient_dot[m][n] = schmidt_norm[n + 1][m] * self.coefficient_dot[m][n]
+
+    def _time_adjust_gauss(self, time):
+        '''Time adjust the Gauss Coefficients
+        
+            There is a very basic cache happening here where if the time delta 
+            hasn't changed then the previous calculation is returned
+        '''
+        current_delta_time = _calculate_decimal_year(time) - self.epoch
+        if self._last_calculated_datetime != current_delta_time:
+            self._update_time_coefficients(current_delta_time)
+            self._last_calculated_datetime = current_delta_time
+        return self.time_adjusted_coefficients_cache
+
+    def _update_time_coefficients(self, delta_time):
+        for n in range(1, self.max_order + 1):
+            for m in range(0, n + 1):
+                self.time_adjusted_coefficients_cache[m][n] = (self.coefficient[m][n] +
+                                                         delta_time * self.coefficient_dot[m][n])
+                if (m != 0):
+                    self.time_adjusted_coefficients_cache[n][m - 1] = (self.coefficient[n][m - 1] +
+                                                                 delta_time * self.coefficient_dot[n][m - 1])
+
+
+class WorldMagneticModel:
+    '''Class for calculating geomagnetic variation according to the world magnetic model
+
+    Example Usage:
+
+    >>> from geomag import WorldMagneticModel
+    >>> WorldMagneticModel().calc_mag_field(80,0).declination
+    -6.1335150785195536
+
+    '''
+    
+    radius_earth = 6371200
+
+    _northerly_intensity = None
+    _easterly_intensity = None
+    _vertical_intensity = None
+    _horizontal_intensity = None
+    _total_intensity = None
+    _declination = None
+    _inclination = None
+    _grid_variation = None
+        
+    def __init__(self, world_magnetic_model_filename=DEFAULT_PATH):
+        '''__init__(self,world_magnetic_model_filename='WMM.COF')
+        Loads a file containing the constants for the magnetic model.
+
+        The coefficients for the model are included for the year 2015 if no
+        variable is provided.
+        File should be in the format of:
+
+        =========    ========    ========    ========    ========    ========
+        degree(n)    order(m)    g |mnt0|    h |mnt0|    g |mnt0|    h |mnt0|
+        =========    ========    ========    ========    ========    ========
+
+        .. |mnt0| replace:: \ :sup:`m`:sub:`n`\ (t\ :sub:`0`\ )
+
+        '''
+        self.data = MageneticModelData(world_magnetic_model_filename)
+        self.k = recursion_constants(self.data.array_size)
+        
```

Comment:
to be PEP8 compatible, all class members should be initialized in the constructor.